app/models/message.rb

```diff
@@ -8,8 +8,17 @@ def to_s
     value
   end
 
+  def self.make_digest(val)
+    # convert from unsinged to signed int64
+    XXhash.xxh64(val) - 9223372036854775808
```

Comment:
What's this doing? I'm not super familiar with the magic ruby does behind the scene for handling integers.

So this returns a hash > 2**63, so you'd need an unsigned int64 type to represent it:

```ruby
XXhash.xxh64("foox")
```

Ruby seems to do the right thing internally right? It uses 8-bytes:

```ruby
2.4.2 :007 > XXhash.xxh64("foox").size
 => 8 
```

Reply:
Oh is the :bigint data type in the database always signed, so when it sees > 2**63 it freaks out?

Reply:
(Although bigint unsigned seems to be available: https://www.philipcurtis.com/bigint-unsigned-ids-in-rails5-0/, the hack here is fine but a better comment would help explain what's going on)

Reply:
Yeah PostgreSQL only supports signed, so I made this trick which seems to be fastest way of doing this in Ruby. If you know better, let me know how to convert it. I think some bitshifting also does the job, but this looks pretty straightforward.

Reply:
I though about it for a wihle. Ruby is a PITA.  I think this would correctly convert from unsigned to signed:

```ruby
-1*(~(value+1) & 0xFFFF_FFFF_FFFF_FFFF)
```
But the subtraction should be faster, and be good enough.